File: SDKs/amazonaws.com/states/2016-11-23/go/pkg/models/operations/listactivities.go
package operations

import (
"openapi/pkg/models/shared")

type ListActivitiesQueryParams struct {
    MaxResults *string `queryParam:"style=form,explode=true,name=maxResults"`
    NextToken *string `queryParam:"style=form,explode=true,name=nextToken"`
    
}


type ListActivitiesXAmzTargetEnum string

const (
    ListActivitiesXAmzTargetEnumAwsStepFunctionsListActivities ListActivitiesXAmzTargetEnum = "AWSStepFunctions.ListActivities"
)


type ListActivitiesHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    XAmzTarget ListActivitiesXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
    
}

type ListActivitiesRequest struct {
    QueryParams ListActivitiesQueryParams 
    Headers ListActivitiesHeaders 
    Request shared.ListActivitiesInput `request:"mediaType=application/json"`
    
}

type ListActivitiesResponse struct {
    ContentType string 
    InvalidToken *interface{} 
    ListActivitiesOutput *shared.ListActivitiesOutput 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/states/2016-11-23/typescript/src/sdk/models/operations/listactivities.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ListActivitiesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=maxResults" })
  maxResults?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=nextToken" })
  nextToken?: string;
}

export enum ListActivitiesXAmzTargetEnum {
    AwsStepFunctionsListActivities = "AWSStepFunctions.ListActivities"
}


export class ListActivitiesHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: ListActivitiesXAmzTargetEnum;
}


export class ListActivitiesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: ListActivitiesQueryParams;

  @SpeakeasyMetadata()
  headers: ListActivitiesHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.ListActivitiesInput;
}


export class ListActivitiesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  invalidToken?: any;

  @SpeakeasyMetadata()
  listActivitiesOutput?: shared.ListActivitiesOutput;

  @SpeakeasyMetadata()
  statusCode: number;
}
